job_executor/model/datastore.py

```diff
@@ -255,10 +257,18 @@ def delete_draft(self, job_id: str, dataset_name: str):
         dataset_operation = self.draft_version.get_dataset_operation(
             dataset_name
         )
-        if not dataset_is_draft:
+        if dataset_operation != "REMOVE" and rollback_remove:
+            log_message = f"{dataset_name} is not scheduled for removal"
+            self._log(job_id, log_message, level="ERROR")
+            job_service.update_job_status(job_id, "failed", log_message)
+            return
+        if (not dataset_is_draft) or (
+            dataset_operation == "REMOVE" and not rollback_remove
+        ):
             log_message = f'Draft not found for dataset name: "{dataset_name}"'
             self._log(job_id, log_message, level="ERROR")
             job_service.update_job_status(job_id, "failed", log_message)
+            return
         else:
```

Comment:
since the ifs are now all guard statements returning if true, this no longer needs to be in an else statement